template/package.json

```diff
@@ -49,73 +49,76 @@
     "karma-mocha": "^1.3.0",
     "karma-phantomjs-launcher": "^1.0.2",
     "karma-phantomjs-shim": "^1.4.0",
     "karma-sinon-chai": "^1.3.1",
     "karma-sourcemap-loader": "^0.3.7",
-    "karma-spec-reporter": "0.0.31",
-    "karma-webpack": "^2.0.2",
-    "mocha": "^3.2.0",
+    "karma-spec-reporter": "0.0.32",
+    "karma-webpack": "^3.0.0",
+    "mocha": "^5.1.1",
     "chai": "^4.1.2",
     "sinon": "^4.0.0",
-    "sinon-chai": "^2.8.0",
-    "inject-loader": "^3.0.0",
-    "babel-plugin-istanbul": "^4.1.1",
+    "sinon-chai": "^3.0.0",
+    "inject-loader": "^4.0.1",
+    "babel-plugin-istanbul": "^4.1.6",
     "phantomjs-prebuilt": "^2.1.14",
     {{/if_eq}}
     {{#e2e}}
-    "babel-register": "^6.22.0",
-    "chromedriver": "^2.27.2",
-    "cross-spawn": "^5.0.1",
-    "nightwatch": "^0.9.12",
-    "selenium-server": "^3.0.1",
+    "@babel/register": "^7.0.0-beta.49",
+    "chromedriver": "^2.38.2",
+    "geckodriver": "^1.11.0",
+    "cross-spawn": "^6.0.5",
+    "nightwatch": "^1.0.3",
+    "selenium-server": "^3.11.0",
     {{/e2e}}
-    "autoprefixer": "^7.1.2",
-    "babel-core": "^6.22.1",
+    "autoprefixer": "^8.3.0",
+    "@babel/core": "^7.0.0-beta.49",
+    "@babel/plugin-syntax-jsx": "^7.0.0-beta.49",
+    "@babel/plugin-transform-runtime": "^7.0.0-beta.49",
+    "@babel/preset-env": "^7.0.0-beta.49",
+    "@babel/preset-stage-2": "^7.0.0-beta.49",
     "babel-helper-vue-jsx-merge-props": "^2.0.3",
-    "babel-loader": "^7.1.1",
-    "babel-plugin-syntax-jsx": "^6.18.0",
-    "babel-plugin-transform-runtime": "^6.22.0",
-    "babel-plugin-transform-vue-jsx": "^3.5.0",
-    "babel-preset-env": "^1.3.2",
-    "babel-preset-stage-2": "^6.22.0",
-    "chalk": "^2.0.1",
+    "babel-loader": "^8.0.0-beta.3",
+    "babel-plugin-transform-vue-jsx": "^4.0.1",
+    "chalk": "^2.4.1",
     "copy-webpack-plugin": "^4.0.1",
     "css-loader": "^0.28.0",
-    "extract-text-webpack-plugin": "^3.0.0",
+    "mini-css-extract-plugin": "^0.4.0",
     "file-loader": "^1.1.4",
     "friendly-errors-webpack-plugin": "^1.6.1",
-    "html-webpack-plugin": "^2.30.1",
+    "html-webpack-plugin": "^3.2.0",
     "webpack-bundle-analyzer": "^2.9.0",
     "node-notifier": "^5.1.2",
     "postcss-import": "^11.0.0",
     "postcss-loader": "^2.0.8",
     "postcss-url": "^7.2.1",
     "semver": "^5.3.0",
-    "shelljs": "^0.7.6",
-    "optimize-css-assets-webpack-plugin": "^3.2.0",
-    "ora": "^1.2.0",
+    "shelljs": "^0.8.1",
+    "optimize-css-assets-webpack-plugin": "^4.0.0",
+    "ora": "^2.0.0",
     "rimraf": "^2.6.0",
     "uglifyjs-webpack-plugin": "^1.1.1",
-    "url-loader": "^0.5.8",
-    "vue-loader": "^13.3.0",
-    "vue-style-loader": "^3.0.1",
+    "url-loader": "^1.0.1",
+    "vue-loader": "^15.0.3",
+    "vue-style-loader": "^4.1.0",
     "vue-template-compiler": "^2.5.2",
     "portfinder": "^1.0.13",
-    "webpack": "^3.6.0",
-    "webpack-dev-server": "^2.9.1",
+    "webpack": "^4.6.0",
+    "webpack-cli": "^3.0.3",
```

Comment:
is `webpack-cli` required for vue template?

Reply:
I'm not entirely sure if `webpack-cli` is *required*, but all the migration guides I looked at said to install it. The CLI code was split out from `webpack` in v4 and I wasn't really sure how to check if the generator or the templates use any of its functionality anywhere. I had planned to look into it further, but never got around to it.